core/primitives/src/shard_layout.rs

```diff
@@ -90,11 +108,62 @@ fn is_top_level_account(top_account: &AccountId, account: &AccountId) -> bool {
         Some(rest) => rest.is_empty() || rest.ends_with("."),
     }
 }
+
+#[derive(Hash, Clone, Debug, Copy, PartialEq, Eq, PartialOrd, Ord)]
+pub struct ShardUId {
+    pub version: ShardVersion,
+    pub shard_id: u32,
+}
+
+impl ShardUId {
+    pub fn default() -> Self {
+        Self { version: 0, shard_id: 0 }
+    }
+    pub fn to_vec(&self) -> [u8; 8] {
```

Comment:
hmm it doesn't actually return a vector?

Reply:
renamed it to `to_bytes`